### PR TITLE
Implement changes required for #265

### PR DESCRIPTION
**Pull request by pro-Nate**
Let me know if there's anything missing.

### CODE REVIEW

File: py/shared/emails/plumbing.py [hunk 1]
```diff
@@ -343,7 +343,7 @@ async def send_emails(
         )
 
     @concurrent('low')  # noqa: C901 (ignore complexity)
-    async def record_email_event(self, raw_message: str):
+    async def record_email_event(self, raw_message: str):  # noqa: C901
```
Comment (samuelcolvin) [inline py/shared/emails/plumbing.py:346]: I doubt this is necessary since the function hasn't changed.

File: py/shared/sql/models.sql [hunk 2]
```diff
@@ -92,6 +92,7 @@ CREATE TABLE events (
   start_ts TIMESTAMPTZ NOT NULL,
   timezone VARCHAR(63) NOT NULL DEFAULT 'Europe/London',
   duration INTERVAL,
+  youtube_video_id VARCHAR(140),
```
Comment (samuelcolvin) [inline py/shared/sql/models.sql:95]: you need to add a patch to `db.py` to add this field.

File: js/src/events/Main.js [hunk 3]
```diff
@@ -15,6 +15,8 @@ import BookEvent from './Book'
 import DonateToEvent from './Donate'
 import WaitingList from './WaitingList'
 import Thanks from '../donations/Thanks'
+import YoutubePlayer from '../general/YoutubePlayer'
```
Comment (samuelcolvin) [inline js/src/events/Main.js:18]: I don't think you've included this file in git which is why CI is failing.

Best to read the errors if CI fails before asking for a review.

File: py/shared/sql/models.sql (hunk 2 above)
Comment (pro-Nate) [reply py/shared/sql/models.sql:95]: Thanks. Done.

File: py/shared/emails/plumbing.py (hunk 1 above)
Comment (pro-Nate) [reply py/shared/emails/plumbing.py:346]: It is. The tests don't run  without it:
`py/shared/emails/plumbing.py:346:5: C901 'BaseEmailActor.record_email_event' is too complex (13)`

File: js/src/events/Main.js (hunk 3 above)
Comment (pro-Nate) [reply js/src/events/Main.js:18]: Dunno what happened there been trying out a graphical git client.
Shall we add failure notifications to travis?

File: js/src/general/YoutubePlayer.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import React from 'react'
+
+
+export const YoutubePlayer = ({vid, ...props}) => (
+    <div {...props}>
+        <div className={`video-wrapper youtube mt-3 mb-4`} style={aspectRatio_16_9_Style} >
```
Comment (samuelcolvin) [inline js/src/general/YoutubePlayer.js:6]:
```suggestion
        <div className="video-wrapper youtube mt-3 mb-4" style={aspectRatio_16_9_Style} >
```

regarding `aspectRatio_16_9_Style` I've tried to put all styles in css unless they have to be truly dynamic, probably best to follow that convention.

File: js/src/events/Dashboard.js [hunk 5]
```diff
@@ -337,6 +337,12 @@ export class EventsDetails extends RenderDetails {
             },
             render: (v, item) => <ImageThumbnail image={v} alt={item.name} image_type="main" width={150}/>,
           },
+          youtube_video_id: {
+            title: 'Youtube video ID',
+            render: (vid) => vid ?
+              <a target="_blank" rel="noopener noreferrer" href={'https://www.youtube.com/watch?v={vid}'}>https://www.youtube.com/watch?v={vid}</a>
```
Comment (samuelcolvin) [inline js/src/events/Dashboard.js:343]:
```suggestion
              <a target="_blank" rel="noopener noreferrer" href={`https://www.youtube.com/watch?v=${vid}`'}>https://www.youtube.com/watch?v={vid}</a>
```

Maybe I'm missing something, but doesn't it need to be like this?

File: js/src/events/Main.js [hunk 6]
```diff
@@ -160,6 +162,7 @@ const EventDetails = WithContext(({ctx, event, uri, ticket_types, existing_ticke
         </div>
       )}
       <h2>About</h2>
+      {event.youtube_video_id && <YoutubePlayer vid={event.youtube_video_id} style={{maxWidth: '680px'}} />}
```
Comment (samuelcolvin) [inline js/src/events/Main.js:165]: again, better where possible to put styles in scss

File: py/shared/emails/plumbing.py (hunk 1 above)
Comment (samuelcolvin) [reply py/shared/emails/plumbing.py:346]: probably because you're using a different version of python or flake8

**Review by samuelcolvin — COMMENTED** [review]
otherwise looking good, though I haven't run it, just read through.

File: js/src/general/YoutubePlayer.js (hunk 4 above)
Comment (pro-Nate) [reply js/src/general/YoutubePlayer.js]: Got it.

File: js/src/events/Dashboard.js (hunk 5 above)
Comment (pro-Nate) [reply js/src/events/Dashboard.js]: JSX parses that but I've changed it to stick to convention

**Comment by samuelcolvin** [comment]
I haven't tried this, but reading. through it looks, great.

LGTM, merge when you're ready.

**Comment by samuelcolvin** [comment]
To run the patch you'll want to run the command

    heroku run './run.py patch <patch name> --live' -a nosht

**Pull request merged**